addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/scanrules/ActiveScriptScanRule.java

```diff
@@ -0,0 +1,247 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.scripts.scanrules;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.ConfigurationUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Plugin;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata;
+import org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadataProvider;
+import org.zaproxy.zap.control.AddOn;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+
+public class ActiveScriptScanRule extends ActiveScriptHelper {
+
+    private static final Logger LOGGER = LogManager.getLogger(ActiveScriptScanRule.class);
+    private static final String SCRIPT_NAME_KEY =
+            "activeScriptScanRuleReflectionWorkAround.script.name";
+    private ExtensionScript extScript;
+    private ScriptWrapper script;
+    private CachedScriptInterfaces cachedScriptInterfaces;
+    private ScanRuleMetadata metadata;
+
+    /** Used via reflection in {@link org.parosproxy.paros.core.scanner.PluginFactory} */
+    public ActiveScriptScanRule() {}
+
+    public ActiveScriptScanRule(ScriptWrapper script, ScanRuleMetadata metadata) {
+        this.script = script;
+        this.cachedScriptInterfaces = new CachedScriptInterfaces(script);
+        this.metadata = metadata;
+    }
+
+    @Override
+    public Configuration getConfig() {
+        var config = super.getConfig();
+        if (config != null && script != null) {
+            // TODO: Do we really need to create a copy here? The intention is to avoid potential
+            //  issues stemming from concurrent modification of the same configuration object.
+            config = ConfigurationUtils.cloneConfiguration(config);
+            config.addProperty(SCRIPT_NAME_KEY, script.getName());
+        }
+        return config;
+    }
+
+    @Override
+    public void init() {
+        super.init();
+        if (script == null) {
+            // This is a workaround for the case when the instance is created via reflection in the
+            // HostProcess class.
```

Comment:
Happy to discuss other possible workarounds or solutions for this problem.

Reply:
We should change core at some point to have factories of scan rules.